modules/platforms/dotnet/Apache.Ignite/Internal/Linq/ResultSelector.cs

```diff
@@ -195,6 +207,65 @@ private static RowReader<T> EmitSingleColumnReader<T>(IColumnMetadata column, Re
         return (RowReader<T>)method.CreateDelegate(typeof(RowReader<T>));
     }
 
+    private static RowReader<T> EmitMemberInitReader<T>(
+        MemberInitCacheTarget target,
+        IReadOnlyList<IColumnMetadata> columns,
+        ResultSelectorOptions options)
+    {
+        var method = new DynamicMethod(
+            name: $"MemberInitFromBinaryTupleReader_{typeof(T).FullName}_{GetNextId()}",
+            returnType: typeof(T),
+            parameterTypes: new[] { typeof(IReadOnlyList<IColumnMetadata>), typeof(BinaryTupleReader).MakeByRefType() },
+            m: typeof(IIgnite).Module,
+            skipVisibility: true);
+
+        var il = method.GetILGenerator();
+        var ctorParams = target.CtorInfo.GetParameters();
+        var membersInitiated = target.PropertiesOrFields;
+
+        if (ctorParams.Length + membersInitiated.Count != columns.Count)
+        {
+            throw new InvalidOperationException("Constructor parameter count + initialized members parameter count" +
```

Comment:
Is this situation possible?

Reply:
AFAIK no but I prefer to keep it as safeguard JIC.